astropy/time/core.py

```diff
@@ -1633,6 +1634,26 @@ def max(self, axis=None, out=None, keepdims=False):
             )
         return self[self._advanced_index(self.argmax(axis), axis, keepdims)]
 
+    def _ptp_impl(self, axis=None, out=None, keepdims=False):
+        if out is not None:
+            raise ValueError(
+                "Since `Time` instances are immutable, ``out`` "
+                "cannot be set to anything but ``None``."
+            )
+        return self.max(axis, keepdims=keepdims) - self.min(axis, keepdims=keepdims)
+
+    if NUMPY_LT_2_0:
+        _ptp_decorator = lambda f: f
+    else:
+        _ptp_decorator = deprecated("6.1", alternative="np.ptp")
```

Comment:
Is `TimeBase.ptp()` deprecated or not?

Reply:
It is deprecated but a warning isn't actionable for an end user who's running numpy 1.x, and we don't want to encourage users to upgrade to numpy 2 yet as it might still break their envs in ways we do not control.

Reply:
Hum, you know, for some reason I thought it could not be available, but I'm really not sure where I got that from, so I guess it's just my own mistake.

Reply:
I do not recall exactly, but I thought the thinking was to treat it a bit like `Quantity`, where `.ptp` is essentially inherited from `ndarray`, so it is simply removed in numpy 2.0. Not sure this was necessarily the best decision...

Reply:
I'd agree on removing the condition on `NUMPY_LT_2_0` in #16583 if that makes sense